lib/web_ui/lib/src/engine/canvaskit/text.dart

```diff
@@ -976,6 +977,16 @@ class CkParagraphBuilder implements ui.ParagraphBuilder {
 
   /// Builds the CkParagraph with the builder and deletes the builder.
   SkParagraph _buildSkParagraph() {
+    if (browserSupportsCanvaskitChromium) {
+      final String text = _paragraphBuilder.getText();
```

Comment:
Why do we need to read the text string back? Don't we already have it?

Now that I'm looking at the Skia API, it feels a little too chatty. I feel like we should be able to provide all this information - text, word breaks, graphemes, line breaks - in a single method call.

But maybe we can revisit this later. If this works, let's keep it. It's not public API anyway. Or maybe I just need some likbez on why this should be this way.

Reply:
For anyone curious: https://en.wikipedia.org/wiki/Likbez

Reply:
Thanks @hterkelsen for the link 😄 I was wondering about that word myself, and thought it could've been a typo.

The reason we need to ask Skia to give us the string is because SkParagraphBuilder inserts extra text into the string. E.g. when we call `addPlaceholder(...)`, SkParagraphBuilder inserts a magical character into the string (and this is an implementation detail that could change or even go away in the future).

A little bit of history to explain how the API became chatty:

Initially, we had a single method called `buildWithClientInfo()` (to be called instead of `build()`) which takes parameters for all the information it needs (i.e. line breaks, word breaks, graphemes, (and bidi at the time)).

Then we started playing with ICU4X for the bidi information. ICU4X operates on UTF8, but browser APIs operate on UTF16. So we needed a way to indicate to Skia whether the array we are passing is UTF8 or UTF16 indices. That's how the `set***Utf8()` and `set***Utf16()` APIs were born.